src/sdk/noobaa_s3_client/noobaa_s3_client.js

```diff
@@ -0,0 +1,148 @@
+function get_s3_client_v3_params(params) {
+    let s3_client;
+
+    // we use config.AWS_SDK_VERSION_3_DISABLED as workaround in case we want to use only AWS SDK V2
+    if (should_use_sdk_v2(params) || config.AWS_SDK_VERSION_3_DISABLED) {
+        change_s3_client_params_to_v2_structure(params);
+        s3_client = new S3ClientSDKV2(params);
+    } else {
+        s3_client = new S3(params);
+    }
+    return s3_client;
+}
+
+// Here we pack all the functions that would help us decide which version of the sdk should we use
+// How we choose the AWS SDK version for s3 Client?
+// 1. signatureVersion defined? if it is 'v2' => AWS SDK V2
+// 2. maxRedirects defined? if it is a non-negative number => AWS SDK V2
+// 3. AWS SDK V3
+function should_use_sdk_v2(params) {
+    if (params.signatureVersion) {
+        const use_sdk_v2_by_signature_version = should_use_sdk_v2_by_signature_version(params.signatureVersion);
+        if (use_sdk_v2_by_signature_version) return use_sdk_v2_by_signature_version;
+    }
+
+    if (params.maxRedirects) {
+        const use_sdk_v2_by_max_redirects = should_use_sdk_v2_by_max_redirects(params.maxRedirects);
+        return use_sdk_v2_by_max_redirects;
+    }
+
+    return false;
+}
+
+// signature version v2 is only supported in AWS SDK V2
+// v2: The signature version to sign requests with (overriding the API configuration): 'v2' or 'v4'
+// v3: Deprecated, only supports signature 'v4'.
+function should_use_sdk_v2_by_signature_version(signature_version) {
+    return signature_version === 'v2';
+}
+
+// max redirects is only supported in AWS SDK V2
+// v2: The maximum amount of redirects to follow for a service request (Integer, default is 10).
+// v3: Deprecated. SDK does not follow redirects to avoid unintentional cross-region requests.
+function should_use_sdk_v2_by_max_redirects(max_redirects) {
+    return max_redirects >= 0;
+}
+
+// we will use this function in debugging to print the class type easily
+function get_sdk_class_str(s3) {
+    if (is_instance_is_S3ClientSDKV2(s3)) return 'S3 Client SDK V2';
+    return 'S3 Client SDK V3';
+}
+
+function is_instance_is_S3ClientSDKV2(s3) {
+    return s3 instanceof S3ClientSDKV2;
+}
+
+// This function gets map of params as they are written in AWS SDK V3
+// and change the map of params as they appear in AWS SDK V2.
+// The comments are based on AWS SDK V3 upgrading notes:
+// https://github.com/aws/aws-sdk-js-v3/blob/main/UPGRADING.md
+// TODO
+function change_s3_client_params_to_v2_structure(params) {
+    // v2: s3ForcePathStyle, v3: forcePathStyle
+    replace_filed(params, 'forcePathStyle', 's3ForcePathStyle');
+
+    // v2: s3DisableBodySigning, v3: applyChecksum
+    replace_filed(params, 'applyChecksum', 's3DisableBodySigning');
+
+    // v2: sslEnabled, v3: tls
+    replace_filed(params, 'tls', 'sslEnabled');
+
+    // v2: maxRetries, v3: maxAttempts.
+    if (params.maxAttempts > 0) {
```

Comment:
do we currently use `maxRetries` anywhere in the code? couldn't find it. since there are a lot of changes, we should only map the ones we currently use